plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -155,8 +156,12 @@ func (c *nodePlugin) admitPod(nodeName string, a admission.Attributes) error {
 		return nil
 
 	case admission.Delete:
-		// get the existing pod
+		// get the existing pod from the server cache
```

Comment:
If you missed in the cache, other weird things are likely to be happening with requests for secrets and the like.  I'm ok doing this here, but it seems like a bandaid on a bullet hole.